lib/spack/spack/cmd/load.py

```diff
@@ -102,16 +101,14 @@ def load(parser, args):
         )
         return 1
 
-    with spack.store.db.read_transaction():
-        if "dependencies" in args.things_to_load:
-            include_roots = "package" in args.things_to_load
-            specs = [
-                dep for spec in specs for dep in spec.traverse(root=include_roots, order="post")
-            ]
+    if args.things_to_load != "package,dependencies":
+        tty.warn(
+            "The `--only` flag in spack load is deprecated and will be removed in Spack v0.21.0"
+        )
```

Comment:
👍

Reply:
Do we need to bump version here?

Reply:
@haampie I think this :point_up: needs to be:
```suggestion
    if args.things_to_load != "package,dependencies":
        tty.warn(
            "The `--only` flag in spack load is deprecated and will be removed in Spack v0.22.0"
        )
```